app/data_model/list_store.py

```diff
@@ -0,0 +1,82 @@
+import random
+from string import ascii_letters
+from collections import OrderedDict
+
+from structlog import get_logger
+
+from app.settings import EQ_LIST_ITEM_ID_LENGTH
+
+logger = get_logger()
+
+
+def random_string(length):
+    return ''.join(random.choice(ascii_letters) for _ in range(length))
+
+
+class ListStore:
+    """ Storage for lists.
+    """
+
+    def __init__(self, existing_items=None):
+        if not existing_items:
+            existing_items = []
+
+        self._lists = OrderedDict(existing_items)
+
+    def __getitem__(self, item):
+        try:
+            return self._lists[item]
+        except KeyError:
+            return list()
+
+    def _generate_identifier(self):
+        """ Generate an unused random 6 character string"""
+        while True:
+            candidate = random_string(EQ_LIST_ITEM_ID_LENGTH)
+            if candidate not in self.list_item_ids():
+                return candidate
+
+    def list_item_ids(self):
+        ids = []
+        for named_list in self._lists.values():
+            ids.extend(named_list)
+
+        return ids
+
+    def delete_list_item_id(self, list_name, item_id):
+        self._lists[list_name].remove(item_id)
+        if not self._lists[list_name]:
+            del self._lists[list_name]
+
+    def add_list_item(self, list_name):
```

Comment:
Could this not be implemented with `__setitem__`?

Reply:
I'm not sure whether __setitem__ makes sense here. 

This method is generating a new identifier and adding it to the named list.
```
new_list_item_id = store.add_list_item('people')
```
So if we used __setitem__ you'd need to split the generation out. That might be a good idea though. I'll have a think